tests/unit/training/lightning/test_default_trainer.py

```diff
@@ -9,3 +21,257 @@ def test_default_trainer():
         experiment_version="x0",
     )
     assert isinstance(result, pl.Trainer)
+
+
+class MockModel(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.linear = torch.nn.Linear(10, 5)
+
+    def forward(self, x):
+        return self.linear(x)
+
+
+@pytest.fixture
+def mock_model():
+    return MockModel()
+
+
+@pytest.fixture
+def mock_trace_input():
+    return torch.randn(1, 10)
+
+
+@pytest.fixture
+def mock_lightning_module():
+    """Fixture providing a mock lightning module."""
+    mock_module = MagicMock()
+    mock_module._modules = {}  # pylint: disable=protected-access
+    return mock_module
+
+
+@pytest.fixture
+def trainer_mocks(mocker):
+    """Fixture providing common mocks for trainer tests."""
+    return {
+        "super_save": mocker.patch.object(pl.Trainer, "save_checkpoint"),
+        "jit_export": mocker.patch(
+            "zetta_utils.training.lightning.trainers.default.jit_trace_export"
+        ),
+        "onnx_export": mocker.patch("zetta_utils.training.lightning.trainers.default.onnx_export"),
+        "is_global_zero": mocker.patch.object(
+            ZettaDefaultTrainer, "is_global_zero", new_callable=PropertyMock
+        ),
+        "lightning_module": mocker.patch.object(
+            ZettaDefaultTrainer, "lightning_module", new_callable=PropertyMock
+        ),
+    }
+
+
+def test_save_checkpoint_calls_exports_when_enabled(
+    trainer_mocks, mock_model, mock_trace_input, mock_lightning_module
+):
+    """Test that save_checkpoint calls export functions when exports are enabled."""
```

Comment:
is this test necessary? isn't it simpler just to add an assert line in the code that checks at least 1 export function is enabled?

Reply:
> That's different. I am OK if the user disables both exports, so that assert would be too restricting, anyway.
Right, I misread the test.

But yea, still seems trivial? Is it really doing much more than a simple line coverage test? (i.e., call the save function & check that the saved file exists, etc.). Making sure the functions are called seems very marginally better.